runtime/vm/createramclass.cpp

```diff
@@ -1614,18 +1629,19 @@ loadSuperClassAndInterfaces(J9VMThread *vmThread, J9ClassLoader *classLoader, J9
 						setCurrentExceptionForBadClass(vmThread, J9ROMCLASS_CLASSNAME(interfaceClass->romClass), J9VMCONSTANTPOOL_JAVALANGINCOMPATIBLECLASSCHANGEERROR);
 						return FALSE;
 					}
-					if (!isROMClassUnsafe) {
+					if (!isExemptFromValidation) {
 						/*
 						 * Failure occurs if
-						 * 1) The interface class not public and does not belong to
+						 * 1) The interface class is not public and does not belong to
 						 * the same package as the class being loaded.
 						 * 2) The interface class is public but the class being loaded
 						 * belongs to a module that doesn't have access to the module that
 						 * owns the interface class.
 						 */
 						bool interfaceIsPublic = J9_ARE_ALL_BITS_SET(interfaceClass->romClass->modifiers, J9AccPublic);
-						if ((!interfaceIsPublic && (packageID != interfaceClass->packageID))
-							|| (interfaceIsPublic && (J9_VISIBILITY_ALLOWED != checkModuleAccess(vmThread, vm, romClass, module, interfaceClass->romClass, interfaceClass->module, interfaceClass->packageID, 0)))
+						if (interfaceIsPublic
+								? (J9_VISIBILITY_ALLOWED != checkModuleAccess(vmThread, vm, romClass, module, interfaceClass->romClass, interfaceClass->module, interfaceClass->packageID, 0))
```

Comment:
Same here - embedding ternary ifs in if statements is a readability concern - please revert this refactoring